specs/lib/HalModuleParser.spec.js

```diff
@@ -807,13 +821,127 @@ describe('HalModuleParser', function () {
 					data: Buffer.from([0x02, 0x00, 0x10, 0x00, 0x00, 0xe0, 0x5f, 0x08, 0x88, 0xe0, 0x5f, 0x08, 0x48, 0xee, 0x3f, 0x02]),
 					length: 16,
 					offset: 8126,
-					type: 2
+					type: 2,
+					dynalibLoadAddress: '85fe088',
+					dynalibStartAddress: '23fee48',
+					moduleStartAddress: '85fe000',
 				},
 				{
 					data: Buffer.from([0x01, 0x00, 0xa, 0x00, 0xff, 0xff, 0x20, 0x00, 0xef, 0xbe]),
 					length: 10,
 					offset: 8142,
-					type: 1
+					type: 1,
+					productId: 32,
+					productVersion: 48879,
+				}
+			]
+		};
+		var parser = new HalModuleParser();
+		parser.parseFile(filename)
+			.then(
+				function (fileInfo) {
+						should(fileInfo).be.ok;
+						should(fileInfo.crc.ok).be.ok;
+						should(fileInfo.suffixInfo).eql(expectedSuffixInfo);
+						done();
+				},
+				function (err) {
+						done(err)
+				}).catch(done);
+	});
+
+	it('should read prefix info from P2 user part with larger module prefix containing extensions', function (done) {
+		var filename = path.join(settings.binaries, 'p2-user-part-prefix-extensions.bin');
+		var expectedPrefixInfo = {
+			moduleStartAddy: '85f3f90',
+			moduleEndAddy: '85ffffc',
+			reserved: 0,
+			moduleFlags: ModuleInfo.Flags.PREFIX_EXTENSIONS,
+			moduleVersion: 6,
+			platformID: 32,
+			moduleFunction: 5,
+			moduleIndex: 1,
+			depModuleFunction: 4,
+			depModuleIndex: 1,
+			depModuleVersion: 5300,
+			dep2ModuleFunction: 0,
+			dep2ModuleIndex: 0,
+			dep2ModuleVersion: 0,
+			prefixSize: 136,
+			extensions: [
```

Comment:
My proposal would be to add this:
```
			assets: [
				{
					hashType: 0,
					hashLength: 32,
					hash: '124c92ed9216ee4619119ec8da1a870c352d06941f6ab490af7d9747daa547e0',
					name: '103_system-part1.bin'
				},
				{
					hashType: 0,
					hashLength: 32,
					hash: '7bb001b06dbbca06e8ed2e5c78c0cbac59a58e08f740ddc6b76a453537a03d2e',
					name: 'unknown.bin'
				}
			],
```

Reply:
So, remove `data` and ignore checking it and just checked parsed fields?